File: railway-deploy.py
#!/usr/bin/env python3
"""
Railway Auto-Deploy Script
Automatisches Deployment des Auto-Alert SaaS Services
"""

import subprocess
import sys
import os
import time

def run_command(cmd, timeout=30):
    """Execute Railway command with timeout"""
    try:
        print(f"🔄 Executing: {cmd}")
        result = subprocess.run(
            cmd, 
            shell=True, 
            capture_output=True, 
            text=True, 
            timeout=timeout,
            cwd="D:\\Claude\\Projects\\auto-alert-saas"
        )
        
        if result.returncode == 0:
            print(f"✅ Success: {result.stdout}")
            return result.stdout
        else:
            print(f"❌ Error: {result.stderr}")
            return None
    except subprocess.TimeoutExpired:
        print(f"⏰ Timeout after {timeout}s")
        return None
    except Exception as e:
        print(f"💥 Exception: {e}")
        return None

def main():
    """Main deployment process"""
    print("🚂 Railway Auto-Deploy für Auto-Alert SaaS")
    print("=" * 50)
    
    # Set API Token
    os.environ["RAILWAY_TOKEN"] = "fbe36775-4296-484e-bd0f-72aa378e01e6"
    
    # 1. Check status
    print("\n📊 1. Checking Railway Status...")
    status = run_command("railway status")
    
    # 2. Set Environment Variables (via file approach)
    print("\n🔑 2. Setting Environment Variables...")
    
    env_vars = {
        "NODE_ENV": "production",
        "PORT": "3001", 
        "WEBHOOK_SECRET": "auto-alert-webhook-secret-2024"
    }
    
    for key, value in env_vars.items():
        cmd = f'railway variables --set "{key}={value}"'
        run_command(cmd, timeout=10)
        time.sleep(1)
    
    # 3. Deploy
    print("\n🚀 3. Starting Deployment...")
    deploy = run_command("railway up --detach", timeout=120)
    
    if deploy:
        print("\n✅ Deployment initiated successfully!")
        print("🔍 Check status with: railway logs")
        print("🌐 Open dashboard: https://railway.app/dashboard")
    else:
        print("\n❌ Deployment failed - check Railway CLI manually")
        
    print("\n🎉 Railway Auto-Deploy Complete!")

if __name__ == "__main__":
    main()

File: railway-deploy.js
#!/usr/bin/env node

/**
 * Railway Auto-Deploy für Auto-Alert SaaS
 * Automatisches Service-Deployment mit Environment Variables
 */

const { spawn, exec } = require('child_process');
const path = require('path');

// Set Railway API Token
process.env.RAILWAY_TOKEN = 'fbe36775-4296-484e-bd0f-72aa378e01e6';

console.log('🚂 Railway Auto-Deploy für Auto-Alert SaaS');
console.log('=' .repeat(50));

async function runCommand(command, args = [], options = {}) {
    return new Promise((resolve, reject) => {
        console.log(`🔄 Executing: ${command} ${args.join(' ')}`);
        
        const process = spawn(command, args, {
            cwd: 'D:\\Claude\\Projects\\auto-alert-saas',
            env: { ...process.env },
            stdio: ['inherit', 'pipe', 'pipe'],
            ...options
        });
        
        let stdout = '';
        let stderr = '';
        
        if (process.stdout) {
            process.stdout.on('data', (data) => {
                const output = data.toString();
                stdout += output;
                console.log(output.trim());
            });
        }
        
        if (process.stderr) {
            process.stderr.on('data', (data) => {
                const output = data.toString();
                stderr += output;
                console.error(output.trim());
            });
        }
        
        process.on('close', (code) => {
            if (code === 0) {
                console.log(`✅ Command completed successfully`);
                resolve(stdout);
            } else {
                console.log(`❌ Command failed with code ${code}`);
                reject(new Error(`Command failed: ${stderr}`));
            }
        });
        
        process.on('error', (error) => {
            console.log(`💥 Command error: ${error.message}`);
            reject(error);
        });
    });
}

async function setEnvironmentVariables() {
    console.log('\n🔑 Setting Environment Variables...');
    
    const envVars = {
        'NODE_ENV': 'production',
        'PORT': '3001',
        'WEBHOOK_SECRET': 'auto-alert-webhook-secret-2024'
    };
    
    for (const [key, value] of Object.entries(envVars)) {
        try {
            console.log(`Setting ${key}=${value}`);
            await runCommand('railway', ['variables', '--set', `${key}=${value}`]);
            
            // Small delay between commands
            await new Promise(resolve => setTimeout(resolve, 1000));
        } catch (error) {
            console.log(`⚠️  Failed to set ${key}: ${error.message}`);
        }
    }
}

async function deployService() {
    console.log('\n🚀 Starting Deployment...');
    
    try {
        // Try to deploy
        await runCommand('railway', ['up', '--detach']);
        console.log('\n✅ Deployment initiated successfully!');
        console.log('🔍 Check status: railway logs');
        console.log('🌐 Dashboard: https://railway.app/dashboard');
        return true;
    } catch (error) {
        console.log(`❌ Deployment failed: ${error.message}`);
        return false;
    }
}

async function main() {
    try {
        // 1. Check current status
        console.log('\n📊 1. Checking Railway Status...');
        await runCommand('railway', ['status']);
        
        // 2. Set environment variables
        await setEnvironmentVariables();
        
        // 3. Deploy
        const success = await deployService();
        
        if (success) {
            console.log('\n🎉 Railway Auto-Deploy Complete!');
        } else {
            console.log('\n⚠️  Deployment had issues - check Railway CLI manually');
        }
        
    } catch (error) {
        console.error(`💥 Main process error: ${error.message}`);
        process.exit(1);
    }
}

// Run main process
main().catch(console.error);